federation/pkg/dnsprovider/providers/aws/route53/rrsets.go

```diff
@@ -0,0 +1,176 @@
+													Region:        aws.String("ResourceRecordSetRegion"),
+						*/
+						ResourceRecords: []*route53.ResourceRecord{
+							{ // Required
+								Value: aws.String(rrset.Rrdatas()[0]), // Required
+							},
+							// More values...
+						},
+						/*
+							SetIdentifier: aws.String("ResourceRecordSetIdentifier"),
+
+						*/
+						TTL: aws.Int64(rrset.Ttl()),
+						/*
+							TrafficPolicyInstanceId: aws.String("TrafficPolicyInstanceId"),
+							Weight:                  aws.Int64(1),
+						*/
+					},
+				},
+				// More values...
+			},
+			Comment: aws.String("Ubernetes federated service"),
+		},
+		HostedZoneId: aws.String(*rrset.(ResourceRecordSet).rrsets.zone.impl.Id), // Required
+	}
+	_, err := service.ChangeResourceRecordSets(input)
+	if err != nil {
+		// Cast err to awserr.Error to get the Code and
+		// Message from an error.
+		return nil, err
+	}
+	return ResourceRecordSet{input.ChangeBatch.Changes[0].ResourceRecordSet, &rrsets}, nil
+}
+
+func (rrsets ResourceRecordSets) Remove(rrset dnsprovider.ResourceRecordSet) error {
+	// TODO: Identical to above except for Action.  Factor out common stuff
+	input := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &route53.ChangeBatch{ // Required
+			Changes: []*route53.Change{ // Required
+				{ // Required
+					Action: aws.String("DELETE"), // Required
+					ResourceRecordSet: &route53.ResourceRecordSet{ // Required
+						Name: aws.String(rrset.Name()),         // Required
+						Type: aws.String(string(rrset.Type())), // Required
+						/*
+							AliasTarget: &route53.AliasTarget{
+								DNSName:              aws.String("DNSName"),    // Required
+								EvaluateTargetHealth: aws.Bool(true),           // Required
+								HostedZoneId:         aws.String("ResourceId"), // Required
+							},
+							Failover: aws.String("ResourceRecordSetFailover"),
+							GeoLocation: &route53.GeoLocation{
+								ContinentCode:   aws.String("GeoLocationContinentCode"),
+								CountryCode:     aws.String("GeoLocationCountryCode"),
+								SubdivisionCode: aws.String("GeoLocationSubdivisionCode"),
+							},
+							HealthCheckId: aws.String("HealthCheckId"),
+							Region:        aws.String("ResourceRecordSetRegion"),
+						*/
+						ResourceRecords: []*route53.ResourceRecord{
+							{ // Required
+								Value: aws.String(rrset.Rrdatas()[0]), // Required
+							},
+							// More values...
+						},
+						/*
+							SetIdentifier: aws.String("ResourceRecordSetIdentifier"),
+
+						*/
+						TTL: aws.Int64(rrset.Ttl()),
+						/*
+							TrafficPolicyInstanceId: aws.String("TrafficPolicyInstanceId"),
+							Weight:                  aws.Int64(1),
+						*/
+					},
+				},
+				// More values...
+			},
+			Comment: aws.String("Ubernetes federated service"),
+		},
+		HostedZoneId: aws.String(*rrset.(ResourceRecordSet).rrsets.zone.impl.Id), // Required
+	}
+	_, err := rrsets.zone.zones.interface_.service.ChangeResourceRecordSets(input)
```

Comment:
Yeah, that's how most of the DNS interfaces seem to work.  To delete, you have to pass a fully matching record set, not just an ID.  Otherwise it fails, by design.  The method is called Change...(), but Action:"DELETE" makes it equivalent to a Delete() method.